pkg/providers/amifamily/types.go

```diff
@@ -102,9 +123,10 @@ type DescribeImageQuery struct {
 func (q DescribeImageQuery) DescribeImagesInput() *ec2.DescribeImagesInput {
 	return &ec2.DescribeImagesInput{
 		// Don't include filters in the Describe Images call as EC2 API doesn't allow empty filters.
-		Filters:    lo.Ternary(len(q.Filters) > 0, q.Filters, nil),
-		Owners:     lo.Ternary(len(q.Owners) > 0, lo.ToSlicePtr(q.Owners), nil),
-		MaxResults: aws.Int64(1000),
+		Filters:           lo.Ternary(len(q.Filters) > 0, q.Filters, nil),
+		Owners:            lo.Ternary(len(q.Owners) > 0, lo.ToSlicePtr(q.Owners), nil),
+		IncludeDeprecated: aws.Bool(true),
```

Comment:
I think we should notify users when Karpenter is launching instances with deprecated AMIs. IMO we should at least have warning logs. I think we should have a mechanism for users to be able to identify nodes that are deprecated. I'm not sure what that mechanism should be, but so option could be adding an annotation to the nodeclaim/node, eventing, nodeclaims status condition, metrics, or even repeatedly logging all nodes that are using deprecated AMIs. I personally like the idea of having an annotation to signal a Karpenter owned instances that is using a deprecated AMI. @ellistarn any opinions here?

Reply:
I am with you that adding an annotation that an ami is deprecated is a good idea on the nodeclaim

Reply:
@engedaam Thoughts on a warning status condition on the NodeClass?

Reply:
> Thoughts on a warning status condition on the NodeClass?

I like the idea of having the warning status condition on the NodeClass. Customers will need to do the leg work of finding the instances using the deprecated  that are owned by the NodeClass themselves, but I don't think that would be a heavy lift for them. 

> Logging/events strike me as spammy, and potentially ignored. Condition will cause a metric, which has nice benefits, too.

I agree with that

Reply:
@shabbskagalwala can you add the status condition on the EC2NodeClass? https://github.com/aws/karpenter-provider-aws/blob/main/pkg/controllers/nodeclass/status/ami.go

Reply:
@engedaam Done. Since the condition statuses can only be either True, False or Unknown, here's an example of when both amis selected through ami selectors are deprecated. 

```
status:
  amis:
  - id: ami-054c1b6d4be926123
    name: amazon-eks-node-1.29
    requirements:
    - key: kubernetes.io/arch
      operator: In
      values:
      - amd64
  - id: ami-0d20e6af81d7ce999
    name: amazon-eks-arm64-node-1.29
    requirements:
    - key: kubernetes.io/arch
      operator: In
      values:
      - arm64
  conditions:
  - lastTransitionTime: "2024-09-09T04:32:55Z"
    message: AMISelector matched deprecated AMIs ami-054c1b6d4be926123, ami-0d20e6af81d7ce999
    reason: AMIsDeprecated
    status: "True"
    type: AMIsReady
  - lastTransitionTime: "2024-09-09T04:32:55Z"
    message: ""
    reason: InstanceProfileReady
    status: "True"
    type: InstanceProfileReady
  - lastTransitionTime: "2024-09-09T04:32:55Z"
    message: ""
    reason: Ready
    status: "True"
    type: Ready
  - lastTransitionTime: "2024-09-09T04:32:55Z"
    message: ""
    reason: SecurityGroupsReady
    status: "True"
    type: SecurityGroupsReady
  - lastTransitionTime: "2024-09-09T04:32:55Z"
    message: ""
    reason: SubnetsReady
    status: "True"
    type: SubnetsReady
```

and only when one ami is deprecated 
```
status:
  amis:
  - id: ami-0b250d0977747b888
    name: amazon-eks-node-1.29
    requirements:
    - key: kubernetes.io/arch
      operator: In
      values:
      - amd64
  - id: ami-0d20e6af81d7ce999
    name: amazon-eks-arm64-node-1.29
    requirements:
    - key: kubernetes.io/arch
      operator: In
      values:
      - arm64
  conditions:
  - lastTransitionTime: "2024-09-09T06:08:41Z"
    message: AMISelector matched deprecated AMIs ami-0d20e6af81d7ce999
    reason: AMIsDeprecated
    status: "True"
    type: AMIsReady
  - lastTransitionTime: "2024-09-09T04:32:55Z"
    message: ""
    reason: InstanceProfileReady
    status: "True"
    type: InstanceProfileReady
  - lastTransitionTime: "2024-09-09T04:32:55Z"
    message: ""
    reason: Ready
    status: "True"
    type: Ready
  - lastTransitionTime: "2024-09-09T04:32:55Z"
    message: ""
    reason: SecurityGroupsReady
    status: "True"
    type: SecurityGroupsReady
  - lastTransitionTime: "2024-09-09T04:32:55Z"
    message: ""
    reason: SubnetsReady
    status: "True"
    type: SubnetsReady
```